integration-tests/deployment/devenv/environment.go

```diff
@@ -0,0 +1,250 @@
+package devenv
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/AlekSi/pointer"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/crypto"
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+	"github.com/stretchr/testify/require"
+	"github.com/subosito/gotenv"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	ctf_config "github.com/smartcontractkit/chainlink-testing-framework/lib/config"
+	ctftestenv "github.com/smartcontractkit/chainlink-testing-framework/lib/docker/test_env"
+	"github.com/smartcontractkit/chainlink-testing-framework/lib/docker/test_env/job_distributor"
+	"github.com/smartcontractkit/chainlink-testing-framework/lib/networks"
+	"github.com/smartcontractkit/chainlink-testing-framework/lib/utils/ptr"
+
+	"github.com/smartcontractkit/chainlink/integration-tests/ccip-tests/testsetups"
+	clclient "github.com/smartcontractkit/chainlink/integration-tests/client"
+	"github.com/smartcontractkit/chainlink/integration-tests/deployment"
+	"github.com/smartcontractkit/chainlink/integration-tests/docker/test_env"
+	tc "github.com/smartcontractkit/chainlink/integration-tests/testconfig"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay"
+)
+
+const (
+	DevEnv = "devenv"
+)
+
+type EnvironmentConfig struct {
+	Chains   []ChainConfig
+	nodeInfo []NodeInfo
+	JDConfig JDConfig
+}
+
+func NewEnvironment(ctx context.Context, lggr logger.Logger, config EnvironmentConfig) (*deployment.Environment, *DON, error) {
+	chains, err := NewChains(lggr, config.Chains)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create chains: %w", err)
+	}
+	offChain, err := NewJDClient(config.JDConfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create JD client: %w", err)
+	}
+
+	jd, ok := offChain.(JobDistributor)
+	if !ok {
+		return nil, nil, fmt.Errorf("offchain client does not implement JobDistributor")
+	}
+	don, err := NewRegisteredDON(ctx, config.nodeInfo, jd)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create registered DON: %w", err)
+	}
+	nodeIDs := don.NodeIds()
+
+	err = don.CreateSupportedChains(ctx, config.Chains)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return &deployment.Environment{
+		Name:     DevEnv,
+		Offchain: offChain,
+		NodeIDs:  nodeIDs,
+		Chains:   chains,
+		Logger:   lggr,
+	}, don, nil
+}
+
+// DeployPrivateChains deploys private chains and returns the chain configs and a function to deploy the Chainlink nodes
+func DeployPrivateChains(t *testing.T) (
```

Comment:
What does "private" mean in this context?

Reply:
Also this function is massive and would benefit from splitting it up into multiple smaller functions to make it easier to understand.